src/Components/Aspire.Azure.AI.Inference/AspireAzureAIInferenceExtensions.cs

```diff
@@ -0,0 +1,178 @@
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(connectionName);
+
+        var settings = new ChatCompletionsClientServiceComponent().AddClient(
+            builder,
+            DefaultConfigSectionName,
+            configureClient,
+            configureClientBuilder,
+            connectionName,
+            serviceKey: null);
+
+        return new AspireChatCompletionsClientBuilder(builder, serviceKey: null, settings.ModelId, settings.DisableTracing);
+    }
+
+    /// <summary>
+    /// Adds a <see cref="ChatCompletionsClient"/> to the application and configures it with the specified settings.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder"/> to add the client to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureClient">An optional callback to configure the <see cref="ChatCompletionsClientSettings"/>.</param>
+    /// <param name="configureClientBuilder">An optional callback to configure the <see cref="IAzureClientBuilder{TClient, TOptions}"/> for the client.</param>
+    /// <returns>An <see cref="AspireChatCompletionsClientBuilder"/> that can be used to further configure the client.</returns>
+    /// <exception cref="InvalidOperationException">Thrown when endpoint is missing from settings.</exception>
+    /// <remarks>
+    /// <para>
+    /// The client is registered as a singleton with a keyed service.
+    /// </para>
+    /// <para>
+    /// Configuration is loaded from the "Aspire:Azure:AI:Inference" section, and can be supplemented with a connection string named after the <paramref name="name"/> parameter.
+    /// </para>
+    /// </remarks>
+    public static AspireChatCompletionsClientBuilder AddKeyedChatCompletionsClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<ChatCompletionsClientSettings>? configureClient = null,
+        Action<IAzureClientBuilder<ChatCompletionsClient, AzureAIInferenceClientOptions>>? configureClientBuilder = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        var settings = new ChatCompletionsClientServiceComponent().AddClient(
+            builder,
+            DefaultConfigSectionName,
+            configureClient,
+            configureClientBuilder,
+            name,
+            serviceKey: name);
+
+        return new AspireChatCompletionsClientBuilder(builder, serviceKey: name, settings.ModelId, settings.DisableTracing);
+    }
+
+    private sealed class ChatCompletionsClientServiceComponent : AzureComponent<ChatCompletionsClientSettings, ChatCompletionsClient, AzureAIInferenceClientOptions>
```

Comment:
Nested classes should either be at the end or in their own file ("Parent.Nested.cs").

Reply:
Is this a process being migrated to in the Aspire repo, as the other files don't do that and I was following their approach.

Reply:
When nested classes interleaved with class API it turns code navigation into a nightmare.

Reply:
It's hard to tell in the browser whether all API below belong to this type or not...

Reply:
Interleaved nested types with parent type API makes navigating the codebase a nightmare.